lib/puppetserver/ca/action/prune.rb

```diff
@@ -106,6 +124,52 @@ def update_pruned_CRL(crl, pkey)
           crl.sign(pkey, OpenSSL::Digest::SHA256.new)
         end
 
+        def prune_using_serial(crl, key, serialnumbers)
+          removed_serials = []
+          revoked_list = crl.revoked
+          @logger.debug("Removing entries in CRL for issuer " \
+            "#{crl.issuer.to_s(OpenSSL::X509::Name::RFC2253)}") if @logger.debug?
+          serialnumbers.each do |serial|
+            if serial.match(/^(?:0[xX])?[A-Fa-f0-9]+$/)
+              revoked_list.delete_if do |revoked|
+                if revoked.serial == OpenSSL::BN.new(serial.hex)
+                  removed_serials.push(serial)
+                  true
+                end
+              end
+            end
+          end
+          crl.revoked = (revoked_list)
+          @logger.debug("Removed these CRL entries : #{removed_serials}") if @logger.debug?
+          return removed_serials.length
+        end
+
+        def prune_using_certname(crl, key, inventory_file, cadir, certnames)
+          serialnumbers = []
+          @logger.debug("Checking inventory file #{inventory_file} for matching cert names") if @logger.debug?
+          File.open(inventory_file).each_line do |line|
```

Comment:
Consider some error handling around this.  Things like inventory.txt having:

- A blank line
- An invalid serial (should we just skip it or expose an error?)
- And invalid certname designation (should always be `/CN=<certname>`
- A line that isn't formatted like we think (e.g. 'This is a bad line')

Reply:
The file can also be missing.

Reply:
Added check for all the above.